sdk/samples/iot/pnp/pnp_protocol.c

```diff
@@ -319,15 +329,17 @@ az_result pnp_process_device_twin_message(
   IOT_SAMPLE_RETURN_IF_FAILED(az_json_reader_init(&jr, twin_message_span, NULL));
   IOT_SAMPLE_RETURN_IF_FAILED(az_json_reader_next_token(&jr));
 
-  if (!is_partial && az_result_failed(json_child_token_move(&jr, iot_hub_twin_desired)))
+  rc = json_child_token_move(&jr, iot_hub_twin_desired);
+  if (!is_partial && az_result_failed(rc))
   {
     IOT_SAMPLE_LOG_ERROR("Failed to get desired property.");
     return AZ_ERROR_UNEXPECTED_CHAR;
   }
 
   copy_jr = jr;
-  if (az_result_failed(json_child_token_move(&copy_jr, iot_hub_twin_desired_version))
-      || az_result_failed(az_json_token_get_int32(&(copy_jr.token), (int32_t*)&version)))
+  az_result rc1 = json_child_token_move(&copy_jr, iot_hub_twin_desired_version);
+  az_result rc2 = az_json_token_get_int32(&(copy_jr.token), (int32_t*)&version);
+  if (az_result_failed(rc1) || az_result_failed(rc2))
```

Comment:
I am guessing this was previously relying on short-circuiting to do the right thing. Without that, the behavior might be broken.

It would be OK to revert this since we aren't inlining the assignment (nor do we use those locals). Up to you no how to verify the correctness here and update if necessary.

Reply:
@danewalton what are your thoughts on this?  I was mixed on pulling out the rc and then checking it in situations like this where we aren't using the rc for anything else.  See above about my note on the vs compiler bug.  If we believe the bug will get addressed, then I'm good with reverting my change in this type of scenario.

Reply:
I'd say revert to keep it simple

Reply:
To confirm on **correctness**, if `json_child_token_move` fails (returns non-AZ_OK result), is it OK to call `az_json_token_get_int32`?

Reply:
> it doesn't make sense to call az_json_token_get_int32 if json_child_token_move fails.

> I'd say revert to keep it simple


OK, good. So reverting this back to what it originally was, wasn't just a simplicity/readability concern, but also correctness.

It's certainly more difficult to review the correctness of samples since they don't have tests :p